### PR TITLE
WIP Renamed all `deferred` to `trickled`

### PR DESCRIPTION
**Pull request by duchainer**
Aims to implement https://github.com/GodotNetworking/network_synchronizer/issues/86

TODO:
 - Test it out properly
 - Check if we should be using the -ed form of the word, or sometimes `trickling` in the comments are in some namings

### CODE REVIEW

File: scene_synchronizer.h [hunk 1]
```diff
@@ -540,7 +540,7 @@ class ClientSynchronizer : public Synchronizer {
 
 	RBSet<EndSyncEvent> sync_end_events;
 
-	struct DeferredSyncInterpolationData {
+	struct trickledSyncInterpolationData {
```
Comment (AndreaCatania) [inline scene_synchronizer.h:543]: This should be `TrickledSyncInterpolationData` instead.

File: net_utilities.h [hunk 2]
```diff
@@ -317,7 +317,7 @@ struct SyncGroup {
 		void update_from(const RealtimeNodeInfo &p_other) {}
 	};
 
-	struct DeferredNodeInfo {
+	struct trickledNodeInfo {
```
Comment (AndreaCatania) [inline net_utilities.h:320]: This should be `TrickledNodeInfo`

**Comment by AndreaCatania** [comment]
This looks good already, just few fixes, then it's good to go.

File: net_utilities.h (hunk 2 above)
Comment (duchainer) [reply net_utilities.h:320]:
```suggestion
	struct TrickledNodeInfo {
```

File: scene_synchronizer.h (hunk 1 above)
Comment (duchainer) [reply scene_synchronizer.h]: Now done in https://github.com/GameNetworking/network_synchronizer/compare/78c8bdb008f93ec9447d130061ed5d8a4678184a..074a9f75b5f01b743d7bda324935eb769d1c374d

**Comment by duchainer** [comment]
> This looks good already, just few fixes, then it's good to go.

It still needs proper testing, but I checked and it should not have affected any godot methods like:
 - set_deferred()
 - call_deferred()
 - etc.
 As we don't call those anyway in NetSync it seems. ^^

**Comment by AndreaCatania** [comment]
Cool thanks :)

**Pull request merged**